Begin untangling the local MSP singleton - Part I

sykesm opened the pull request.

sykesm committed:
msp: construct DeserializersManager  with local MSP

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
msp/mgmt: move DeserializersManager its user

The interface and implementation are simple helpers around the existing
LocalMSP and are only used in the gossip "message crypto service".

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
Remove unused policy checker from lscc

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
msp/mgmt: move MSPPrincipalGetter to policy

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
msp/mgmt: move TestLocalMSP into mgmt_test

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
peer: stop using GetLocalSigningIdentityOrPanic

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
msp/mgmt: remove GetLocalSigningIdentityOrPanic

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
msp/mgmt: remove LoadLocalMspWithType

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
msp/mgmt: remove LoadLocalMsp

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
msp/mgmt: remove Setup/Deserialize wrappers

The single check for `up` that's done in DeserializeIdentity can be done
in the base implementation.

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

mastersingh24 commented:
This is looking good.  I like the incremental approach to removing some of the gorpy stuff first

sykesm committed:
peer: remove NewConfigSupport and friends

The code is wired but unused and untested.

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
cscc: stop wiring policy.PolicyChecker

The reference is unused so stop passing it and cleanup the references.

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
common: remove the unused common/config package

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
peer: use MSPManager from bundle instead of mspmgmt

This begins to replace calls to mgmt.GetManagerForChain with references
to the channel's MSPManager.

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
peer: remove GetManagerFromChain from prod code

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
lscc: remove GetManagerForChain from prod code

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

sykesm committed:
msp/mgmt: remove reflect based type assertion

Signed-off-by: Matthew Sykes <sykesmat@us.ibm.com>

jyellick reviewed (approved):
Definitely an improvement
